### ISSUE TITLE
Numerical error in generating MBHB

### ISSUE THREAD

**Issue by forXuyx** [mikekatz04/lisa-on-gpu#4]
Hello! First of all thank you for your work! I have some problems generating MBHB signal. I use Pycbc to generate time domain waveforms and subsequently conduct them through tdi. The parameter space I use is as follows:
`    Mc = np.random.uniform(1e6, 1e8)
    q = np.random.uniform(0.01, 1.)
    spin1z = np.random.uniform(-0.99, 0.99)
    spin2z = np.random.uniform(-0.99, 0.99)
    phic = np.random.uniform(0, 2.*np.pi)
    inc = np.random.uniform(0, np.pi)
    psi = np.random.uniform(0, np.pi)
    lambda_ = np.random.uniform(0, 2.*np.pi)
    beta = np.random.uniform(-np.pi/2, np.pi/2)
`
My target length is 320000s of data and the sampling frequency is 0.1Hz. I got the waveform as shown in the picture, with an obvious numerical error. I tried using interpolation methods to smooth hp and hc as much as possible, and I also made the generated data as long as possible or increased the sampling frequency, but it didn't work. If you have time, I look forward to your answer. I wish you good luck with your research!
![image](https://github.com/mikekatz04/lisa-on-gpu/assets/96722266/818ffd83-2aa2-4d1c-b0ec-c483065b78cf)

**Comment by mikekatz04** [comment]
Hello, have you solved this? I am not sure exactly what would be happening. Depending how far after the merger it is, you could clip it. The first place to look is anything that is off in the source waveform. Let me know.

**Comment by forXuyx** [comment]
Hello, sorry for taking so long to reply to you. At present, this problem has been solved. We performed a truncation operation and clipped the wrong waveform. We consider that the reason for this phenomenon is that ‘SEOBNRv4’ is not smooth enough. A frequency domain-based model like ‘IMRPhenom’ might be better? Finally, thank you again for your great work, it really made it easier for me to generate data.
